osbrain/tests/test_agent_transport.py

```diff
@@ -24,18 +25,19 @@ def test_agent_bind_transport_global(nsproxy):
     # Default transport
     agent = run_agent('a0')
     address = agent.bind('PUSH')
-    assert address.transport == 'ipc'
+    assert address.transport == osbrain.config['TRANSPORT']
 
     # Changing default global transport
     osbrain.config['TRANSPORT'] = 'tcp'
     agent = run_agent('a1')
     address = agent.bind('PUSH')
     assert address.transport == 'tcp'
 
-    osbrain.config['TRANSPORT'] = 'ipc'
-    agent = run_agent('a2')
-    address = agent.bind('PUSH')
-    assert address.transport == 'ipc'
+    if os.name == 'posix':
+        osbrain.config['TRANSPORT'] = 'ipc'
+        agent = run_agent('a2')
+        address = agent.bind('PUSH')
+        assert address.transport == 'ipc'
```

Comment:
I think there should probably be a test for global transport configuration, in example:

```python
def test_agent_bind_transport_platform_default(nsproxy):
    """
    Default transport is platform-dependent.
    """
    agent = run_agent('a0')
    address = agent.bind('PUSH')
    if os.name == 'posix':
        assert address.transport == 'ipc'
    else:
        assert address.transport == 'tcp'
```

And then another for the global setting:

```python
def test_agent_bind_transport_global(nsproxy):
    """
    Test global default transport change.
    """
    # Default transport is not `inproc`
    agent = run_agent('a0')
    address = agent.bind('PUSH')
    assert address.transport != 'inproc'

    # Changing default global transport to `inproc`
    osbrain.config['TRANSPORT'] = 'inproc'
    agent = run_agent('a1')
    address = agent.bind('PUSH')
    assert address.transport == 'inproc'
```

Reply:
That reminds me that changing values in the global config does not work on windows, for the same reason I explained earlier when I was talking about the linger thing.
When calling `run_agent()` a new process will be created, but because of the way Windows creates child processes, the values in `config` won't be the ones we just set, but rather the default ones.

I'll add these two tests and try to find a way to make the second one work on Windows.

Reply:
As I mention bellow, I'd leave that until we merge this PR. Just having AppVeyor integrated, even if we are skipping some tests in Windows, is worth it to merge. Then we can see how to gradually reduce the number of skipped tests.

Reply:
You forgot to change this (i.e.: create 2 separate tests). :wink: